fastify.js

```diff
@@ -713,7 +724,7 @@ function fastify (options) {
       if (err) {
         if (frameworkErrors) {
           const id = genReqId(req)
-          const childLogger = logger.child({ reqId: id })
+          const childLogger = createChildLogger(onBadUrlContext, logger, req, id)
```

Comment:
It was because the `badUrl` is handled differently than other cases. For instance, it creates the `Request` and `Reply` objects if the `frameworkErrors` hanlder is passed when creating the fastify app.

It shouldn't have negative impact, but we can ask @mcollina for a second pair of eyes to verify